aviary/mission/gasp_based/phases/groundroll_phase.py

```diff
@@ -1,106 +1,183 @@
+            lower=mass_lower,
+            upper=mass_upper,
+            units="lbm",
+            rate_source=Dynamic.Mission.FUEL_FLOW_RATE_NEGATIVE_TOTAL,
+            ref=mass_ref,
+            defect_ref=mass_defect_ref,
+            ref0=mass_ref0,
+        )
+
+        phase.add_state(
+            Dynamic.Mission.DISTANCE,
+            fix_initial=fix_initial,
+            fix_final=False,
+            lower=distance_lower,
+            upper=distance_upper,
+            units="ft",
+            rate_source="distance_rate",
+            ref=distance_ref,
+            defect_ref=distance_defect_ref,
+            ref0=distance_ref0,
+        )
+
+        phase.add_parameter("t_init_gear", units="s",
+                            static_target=True, opt=False, val=100)
+        phase.add_parameter("t_init_flaps", units="s",
+                            static_target=True, opt=False, val=100)
+
+        # boundary/path constraints + controls
+        # the final TAS is constrained externally to define the transition to the rotation
+        # phase
+
+        phase.add_timeseries_output(Dynamic.Mission.THRUST_TOTAL, units="lbf")
+
+        phase.add_timeseries_output("normal_force")
+        phase.add_timeseries_output(Dynamic.Mission.MACH)
+        phase.add_timeseries_output("EAS", units="kn")
+
+        phase.add_timeseries_output(Dynamic.Mission.LIFT)
+        phase.add_timeseries_output("CL")
+        phase.add_timeseries_output("CD")
+        phase.add_timeseries_output("fuselage_pitch", output_name="theta", units="deg")
+
+        return phase
+
 
+# Adding metadata for the GroundrollPhase
+GroundrollPhase._add_meta_data('fix_initial', val=True)
+GroundrollPhase._add_meta_data('fix_initial_mass', val=False)
+GroundrollPhase._add_meta_data('connect_initial_mass', val=True)
+GroundrollPhase._add_meta_data('duration_bounds', val=(1, 100), units='s')
+GroundrollPhase._add_meta_data('duration_ref', val=1, units='s')
+GroundrollPhase._add_meta_data('TAS_lower', val=0, units='kn')
+GroundrollPhase._add_meta_data('TAS_upper', val=1000, units='kn')
+GroundrollPhase._add_meta_data('TAS_ref', val=100, units='kn')
+GroundrollPhase._add_meta_data('TAS_ref0', val=0, units='kn')
+GroundrollPhase._add_meta_data('TAS_defect_ref', val=None, units='kn')
+GroundrollPhase._add_meta_data('mass_lower', val=0, units='lbm')
+GroundrollPhase._add_meta_data('mass_upper', val=200_000, units='lbm')
+GroundrollPhase._add_meta_data('mass_ref', val=100_000, units='lbm')
+GroundrollPhase._add_meta_data('mass_ref0', val=0, units='lbm')
+GroundrollPhase._add_meta_data('mass_defect_ref', val=100, units='lbm')
+GroundrollPhase._add_meta_data('distance_lower', val=0, units='ft')
+GroundrollPhase._add_meta_data('distance_upper', val=4000, units='ft')
+GroundrollPhase._add_meta_data('distance_ref', val=3000, units='ft')
+GroundrollPhase._add_meta_data('distance_ref0', val=0, units='ft')
+GroundrollPhase._add_meta_data('distance_defect_ref', val=3000, units='ft')
```

Comment:
I think defect_ref needs some special processing.

In dymos, if you don't specify a `defect_ref`, then it uses the value in `ref`.  In Aviary, if we are giving all the defect_refs a default value, then a user might change `ref` and not realize that they need to change `defect_ref` for that ref to be fully propagated.  Might be a conversation to have with Rob. I rarely use different values for ref and defect_ref, but I can see that in aviary, some of the values are different.

Reply:
Agreed, Ken. Here I kept the same behavior from existing Aviary on purpose.

Let's discuss this more. I want to get Aviary to a point where the user doesn't have to think about `defect_ref`s because they really shouldn't have to. We had those here because the GASP mission  optimization problem is very challenging to pose well. I'd much rather not expose users to defect_refs at all at this level.

Do you want to see a change here in this PR?

Reply:
I think we can leave this to be a follow-on discussion.